### DISCUSSION TITLE
Midi CTAG TBD -> TBD Black Box Audio

### DISCUSSION THREAD

**Discussion by ctag-fh-kiel** [ctag-fh-kiel/ctag-tbd#71] [Announcements]
A little teaser of the upcoming MIDI version of the TBD aka. TBD Black Box Audio (BBA) can be seen here:
https://youtu.be/Lrzo0KT3fXU

Thx to wgd modular :)

**Comment by machinehistories** [comment]
Looks amazing. Is this going to be sold as a product or is this meant to be built as shown in the video and if so can you please share a link to the build files for tbd bba. Incredible work!

**Reply by ctag-fh-kiel** [reply]
Hi there, we're still working on a release version, it will still take some time...

**Reply by machinehistories** [reply]
No problem thanks. I wanted to make a feature request for the eurorack module. Essentially it adds the option to use a button to interact with a list. For example, when using the mi rings based algorithm you can choose cv 0-1 or pot 0-1 to select the mode. There are many parameters on many of the various algorithms that act more like lists than analog voltages and it would be cool to add button 0-1 as a control option for these list like parameters. Then when playing the module, for each button press, it simply iterates through the list. I would try to code this but I'm not even sure which file or files need to be edited to implement this feature. This tbd project is seriously developed and it is a little intimidating for someone with only arduino experience. I would still love to try. I think this feature would be really helpful because it could enable cv to be used on more appropriate parameters while still enabling the user to be more hands on interacting with the module on parameters that act more like lists.

**Reply by ctag-fh-kiel** [reply]
Hi there, if you are interested, we can schedule a v-con to do a code walk and I could explain you the sites of interested, where you could try to implement this select/toggle behaviour...
Let me know:)

**Comment by machinehistories** [comment]
That would be terrific and very generous of you. I am available most
anytime except on Wednesday. thanks again

**Comment by schrab** [comment]
What about MIDI-host implementaton, is it ESP32-S3 based?

**Reply by ctag-fh-kiel** [reply]
We plan to include MIDI host, but probably not with the S3. I think we'll use another chip for it. 
Do you know if the MIDI host implementation for the S3 is stable and available?

**Reply by schrab** [reply]
Works on S2/S3 with some issues (no hotplug for example)
Made a quick test with Nanokontrol2 without any issues.

**Comment by machinehistories** [comment]
That's so great. I have a elektra one and I think it is a fantastic midi
controller platform with a well developed sandbox to customize the device.
I feel like that paired up with the new tbd is going to be tremendous.

**Comment by schrab** [comment]
@ctag-fh-kiel one more question, Robert. Can BBA run on ESP32? I have an Audio Kit board with 8MB PSRAM / ES8388 codec but there's a lot of issues with build process. Or it is only ESP32S3 compatible?

**Reply by ctag-fh-kiel** [reply]
It should in principle also work on the ESP32 but you may have to remove / comment out the sections of the USB MIDI support. Also the pin mapping would have to be adapted. I have implemented an ES8388 driver, so you could recycle it. There is already an option for it with idf.py menuconfig, however, I think you will have to make manual adaptations to the code.
Let me know how it goes :)

**Reply by schrab** [reply]
3 days gone. First I've started with BBA sdkconfig, set codec to 8388 in conf / removed all usb related routine / remapped gpio / corrected a lot of compilation errors (4.4->5.1 related I suppose),  but no luck wit boot. Maybe because of initial BBA config made for ESP32S3 SoC.
Now I trying go from another side. Take V2 as base, removed ADC/TRIG/SPI stuff, added MDI and remaking codec.cpp for es8388. Still some bugs with I2C but at least it boots up. 
I'm using IDF v5.1.4, is it fine or I have to use the exact version you compilled with?

UPD: That was wrong UART in Midi.cpp, now it's working fine, just have to set up analog input and gains.

**Comment by ctag-fh-kiel** [comment]
IDF v5.1.4 is fine, that's the version I use

schrab ***@***.***> schrieb am Do., 24. Okt. 2024, 03:51:

**Comment by Cinezaster** [comment]
Any news on the BBA? Still interested.

**Reply by ctag-fh-kiel** [reply]
Hi there, yes, we are in very active development using the newest Espressif Chip (esp32-p4). 
Are you a developer? You could possibly join the team at this early stage :)

**Reply by Cinezaster** [reply]
Yes I'm a developer.
Would be nice to help out.
Not sure if I can dedicate a lot of time, but for sure can help testing.
Where should I start? With the simulator?

**Reply by ctag-fh-kiel** [reply]
Sounds great, I just sent you an email... :)
